src/camera.cpp

```diff
@@ -551,7 +551,7 @@ void Camera::updateViewingRange()
 	f32 viewing_range = g_settings->getFloat("viewing_range");
 	f32 near_plane = g_settings->getFloat("near_plane");
 	m_draw_control.wanted_range = viewing_range;
-	m_cameranode->setNearValue(near_plane * BS);
+	m_cameranode->setNearValue(MYMIN(MYMAX(0, near_plane), 0.5) * BS);
```

Comment:
We have functions for this, either `rangelim(near_plane, 0.0f, 0.5f)` or maybe `std::clamp(...)` not sure which.